src/providers/twitch/eventsub/MessageHandlers.cpp

```diff
@@ -92,8 +92,53 @@ void handleModerateMessage(
         builder.emplaceSystemTextAndUpdate(action.reason.qt(), text);
     }
 
-    builder->messageText = text;
-    builder->searchText = text;
+    builder.setMessageAndSearchText(text);
+    builder->timeoutUser = action.userLogin.qt();
+
+    auto msg = builder.release();
+    runInGuiThread([chan, msg, time] {
+        chan->addOrReplaceTimeout(msg, time);
+    });
+}
```

Comment:
This diff looks goofy, because I used `builder.setMessageAndSearchText(text);` here (that's the only change in that method).